### PR TITLE
Add timestamp as option to DateTime Fields

### PR DESCRIPTION
**Pull request by vanHoi**
This PR adds the possibility to load and dump to timestamps.

This was discussed before in issue #612 and then partly implemented in #1003. Since the branch was stale (still on Marshmellow 2) I decided to start over.

The difficulty with timestamps is that you can't infer a timezone from a timestamp. Since the introduction of AwareDateTime and NaiveDateTime however, this issue is mostly solved. By specifying a default timezone, the user can add the correct timezone when it is required.

### CODE REVIEW

**Comment by lafrech** [comment]
Thanks for the PR and sorry about the response delay.

We didn't get the time to review thoroughly yet, but this looks good.

**Review by georgebike — APPROVED** [review]

File: src/marshmallow/utils.py
Comment (lafrech) [inline src/marshmallow/utils.py]: I'd do it the other way around.

```py
    if not is_aware(value):
        # When a date is naive, use UTC as zone info to prevent using system timezone.
        value = value.replace(tzinfo=dt.timezone.utc)
    return value.timestamp()
```

File: src/marshmallow/fields.py
Comment (lafrech) [inline src/marshmallow/fields.py]: Slightly outdated.

File: src/marshmallow/fields.py
Comment (lafrech) [inline src/marshmallow/fields.py:1221]: Wondering (out loud) about the choices and their naming. I don't use timestamps. Are those two the two common uses?

`timestamp` is a number of seconds. Should we call it `timestamp_s`? I don't think so. Why the need for `timestamp_ms` (and not other units)? Is it also commonly used?

**Review by lafrech — COMMENTED** [review]
Thanks. Looks good.

Minor remarks / questions.

File: src/marshmallow/fields.py
Comment (vanHoi) [reply src/marshmallow/fields.py:1221]: Thanks for the review! Good question. These are the two common use cases (to my knowledge). JavaScript tends to use timestamps in milliseconds, while Python uses seconds by default.

File: src/marshmallow/fields.py
Comment (vanHoi) [reply src/marshmallow/fields.py]: Updated to 3.19 :sparkles:

File: src/marshmallow/utils.py
Comment (vanHoi) [reply src/marshmallow/utils.py]: I agree this is a bit cleaner :heavy_check_mark:

**Review by lafrech — APPROVED** [review]
Let's keep `timestamp` for seconds. We live in Python world. OK for providing `timestamp_ms` even if it opens the door to possibly N other unit requests (µs, ns,...), N should be small anyway.

Giving a little time for another maintainer to review before merging.

**Review by sloria — APPROVED** [review]

**Pull request merged**
